contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -157,17 +156,16 @@ library BytecodeStorage {
             // (^) Note: this value must be adjusted if portions of the selfdestruct purge logic are adjusted.               //
             //---------------------------------------------------------------------------------------------------------------//
             //
-            // store the deploying-contract's address (to be used to gate and call `selfdestruct`)
+            // store the deploying-contract's address (to be used to gate and call `selfdestruct`),
+            // with expected 0-padding to fit a 20-byte address into a 30-byte slot.
             //
             // note: it is important that this address is the executing contract's address
             //      (the address that represents the client-application smart contract of this library)
             //      which means that it is the responsibility of the client-application smart contract
             //      to determine how deletes are gated (or if they are exposed at all) as it is only
             //      this contract that will be able to call `purgeBytecode` as the `CALLER` that is
             //      checked above (op-code 0x33).
-            //
-            // also note: abi.encodePacked will not `0`-pad the address to 32 bytes,
-            //            making the address 20 bytes instead
+            hex"00_00_00_00_00_00_00_00_00_00_00_00", // left-pad 20-byte address with 12 0x00 bytes
             address(this),
```

Comment:
😬 I'm going to take you up on the offer to decline and reject the suggestion, as personally this is much less intuitive/explicit to me.

The suggestion here forces you to have more context re: the underlying higher level language constructs/assumptions, vs. staying in "byte land" and I like being able to avoid the frequency of context switching between direct-bytes and higher-level-languages with code like this (e.g. I like the property of being able to put my brain in bytemode and leave it there for as long as possible before switching). 😅 

That said, if you feel more strongly than this comment suggestions, we can def revisit ser and push a lil update.

Reply:
Your reasoning is definitely good with me, totally fine leaving this as-is!